File: Week 9/Lab/Comparing Two Files (Java)/SameContent.java
// Q4-W9: Comparing Two Files (Java)

import java.io.*;
import java.util.*;

public class SameContent {

    public static boolean sameContent(String filename1, String filename2) {
        String content1 = "", content2 = "";
        BufferedReader br;

        try {
            br = new BufferedReader(new FileReader(filename1));
            while (br.ready()) {
                content1 += (br.readLine() + System.lineSeparator());
            }
            br.close();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }

        try {
            br = new BufferedReader(new FileReader(filename2));
            while (br.ready()) {
                content2 += (br.readLine() + System.lineSeparator());
            }
            br.close();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return content1.equals(content2);
    }

    public static void main(String[] args) {
        if (sameContent(args[0], args[1])) {
            System.out.println("The two files have the same content.");
        }
        else {
            System.out.println("The two files do not have the same content.");
        }
    }
}

File: Week 9/Lab/Comparing Two Files (Python)/same_content.py
# Q5-W9: Comparing Two Files (Python)

import sys


def same_content(file1, file2) -> bool:
    f1 = open(file1, "r")
    f2 = open(file2, "r")
    buffer1 = f1.read()
    buffer2 = f2.read()
    f1.close()
    f2.close()
    return buffer1 == buffer2


if __name__ == "__main__":
    if same_content(sys.argv[1], sys.argv[2]):
        print("The two files have the same content.")
    else:
        print("The two files do not have the same content.")
